packages/runtime/src/account.ts

```diff
@@ -184,6 +184,58 @@ export class StoreAccount implements StoreAccountI {
     return asset.definitions;
   }
 
+  /**
+   * Modifies Asset fields
+   * @param assetId Asset Index
+   * @param fields Fields for modification
+   */
+  modifyAsset (assetId: number, fields: AssetModFields): void {
+    const asset = this.getAssetDef(assetId);
+    if (asset === undefined) {
+      throw new TealError(ERRORS.ASA.ASSET_NOT_FOUND, { assetId: assetId });
+    }
+    // check for blank fields
+    if ((fields.reserve !== "" && asset.reserve === "") || (fields.freeze !== "" && asset.freeze === "") ||
+      (fields.clawback !== "" && asset.clawback === "")) {
+      throw new Error("Cannot reset a blank address");
+    }
+
+    asset.manager = fields.manager;
+    asset.reserve = fields.reserve;
+    asset.freeze = fields.freeze;
+    asset.clawback = fields.clawback;
+  }
+
+  /**
+   * Freeze asset
+   * @param assetId Asset Index
+   */
+  freezeAsset (assetId: number, state: boolean): void {
+    const holding = this.assets.get(assetId);
+    if (holding === undefined) {
+      throw new TealError(ERRORS.ASA.ASSET_HOLDING_NOT_FOUND,
+        { address: this.address, assetId: assetId });
+    }
+    holding["is-frozen"] = state;
```

Comment:
What happens if asset is already frozen?  Is an error thrown or does it has no effect?
Also, could we rename this function to something else, `freezeAsset` suggests `is-frozen` is always true (i.e we are 'freezing' the asset). Maybe `setFreezeState()` ?

Reply:
no effect, yeah setFreezeState is better, algorand uses makeFreezeAsset though